Selenium timeouts in Travis

😢 

I got a seemingly random failure in `test_selenium.py` during testing of #879 (which made no frontend changes) in a TravisCI build.

```txt
frontend/tests/test_selenium.py:185: in wait_for
    return self.fail(err)
E   AssertionError: Timeout waiting for are_results_loaded
```

It just failed again in #885 😭 

```txt
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
frontend/tests/test_selenium.py:185: in wait_for
    return self.fail(err)
E   AssertionError: Timeout waiting for are_results_loaded
---------------------------- Captured stdout setup -----------------------------
capabilities: {'tunnel-identifier': '2973.1', 'public': 'public', 'platform': 'ANY', 'browserName': 'internet explorer', 'version': '11.103', 'name': 'CALC', 'javascriptEnabled': True}
----------------------------- Captured stdout call -----------------------------
loading URL: http://127.0.0.1:8000/tests/
=============== 1 failed, 336 passed, 3 skipped in 60.07 seconds ===============
The command "py.test --ignore=frontend/tests/test_selenium.py --cov" exited with 1.
```

--- a/frontend/tests/test_selenium.py
+++ b/frontend/tests/test_selenium.py
@@ -161,11 +161,21 @@ class SeleniumTestCase(LiveServerTestCase):
         self.driver.set_window_size(*self.window_size)
         super().setUp()
 
+        if WD_IS_USING_SAUCE_LABS and not hasattr(self, '_gateway_ensured'):
+            self.ensure_gateway_works()
+            self._gateway_ensured = True
+
     def tearDown(self):
         if WD_IS_USING_SAUCE_LABS:
             print("Results: http://saucelabs.com/jobs/{}".format(
                 self.driver.session_id
             ))
+
+    def ensure_gateway_works(self):
+        def about():
+            self.load('/about/')
+            return 'CALC' in self.driver.page_source
+        self.wait_for(about)
 
     def load(self, uri='/'):
         url = self.base_url + uri

